From 678d4c9a67a36505eea6598b61429a259787e5f5 Mon Sep 17 00:00:00 2001
From: Vlad Pelin <vlad.pelin@nxp.com>
Date: Wed, 1 Jun 2022 19:50:03 +0300
Subject: [PATCH 50/50] secboot: add placeholder for signature in fip

added new update_fip_cert to account for fip signature

Issue: ALB-8660
Upstream-Status: Pending 

Signed-off-by: Vlad Pelin <vlad.pelin@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/s32cc.dtsi            |  1 +
 plat/nxp/s32/s32_common.mk | 13 +++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/fdts/s32cc.dtsi b/fdts/s32cc.dtsi
index d0c24308f..544b697f7 100644
--- a/fdts/s32cc.dtsi
+++ b/fdts/s32cc.dtsi
@@ -58,6 +58,7 @@
 		};
 
 		hse_reserved: shm@84000000 {
+			compatible = "nxp,s32cc-hse-rmem";
 			reg = <0x0 0x84000000 0x0 0x80000>;	/* 512k */
 			no-map;
 		};
diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index d1ca8f1b9..f2c6d3f77 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -229,6 +229,10 @@ define update_fip
 ${FIPTOOL} update --align ${FIP_ALIGN} --tb-fw $1 --soc-fw-config $2 $3
 endef
 
+define update_fip_cert
+${FIPTOOL} update --align ${FIP_ALIGN} --tb-fw $1 --soc-fw-config $2 --tb-fw-cert $3 $4
+endef
+
 define get_fip_hdr_size
 printf "0x%x" $$(${FIPTOOL} info $1 | awk -F'[=,]' '{print strtonum($$2)}' | sort -n | head -n1)
 endef
@@ -263,7 +267,11 @@ ${DUMMY_FIP}: fiptool ${DUMMY_STAGE} | ${BUILD_PLAT}
 	${Q}${ECHO} "  FIP     $@"
 	${Q}ARGS=$$(echo "${FIP_ARGS}" | sed "s#\(--[^ a]\+\)\s\+\([^ ]\+\)#\1 ${DUMMY_STAGE}#g"); \
 		${FIPTOOL} create $${ARGS} "$@_temp"
+ifneq (${HSE_SECBOOT},)
+	${Q}$(call update_fip_cert, ${DUMMY_STAGE}, ${DUMMY_STAGE}, ${DUMMY_STAGE}, "$@_temp")
+else
 	${Q}$(call update_fip, ${DUMMY_STAGE}, ${DUMMY_STAGE}, "$@_temp")
+endif
 	${Q}mv "$@_temp" $@
 
 ${DUMMY_FIP_S32}: ${DUMMY_FIP}
@@ -388,7 +396,12 @@ FIP_ALIGN := 16
 all: add_to_fip
 add_to_fip: fip ${BL2_W_DTB}
 	$(eval FIP_MAXIMUM_SIZE_10 = $(shell printf "%d\n" ${FIP_MAXIMUM_SIZE}))
+ifneq (${HSE_SECBOOT},)
+	@dd if=/dev/urandom of=${BUILD_PLAT}/dummy_cert bs=1 count=256
+	${Q}$(call update_fip_cert, ${BUILD_PLAT}/bl2_w_dtb.bin, ${BUILD_PLAT}/fdts/${DTB_FILE_NAME}, ${BUILD_PLAT}/dummy_cert, ${BUILD_PLAT}/${FIP_NAME})
+else
 	${Q}$(call update_fip, ${BUILD_PLAT}/bl2_w_dtb.bin, ${BUILD_PLAT}/fdts/${DTB_FILE_NAME}, ${BUILD_PLAT}/${FIP_NAME})
+endif
 	${Q}${ECHO} "Added BL2 and DTB to ${BUILD_PLAT}/${FIP_NAME} successfully"
 	${Q}${FIPTOOL} info ${BUILD_PLAT}/${FIP_NAME}
 	$(eval ACTUAL_FIP_SIZE = $(shell \
-- 
2.17.1

